components/post_view/post_info/post_info.tsx

```diff
@@ -202,7 +209,7 @@ export default class PostInfo extends React.PureComponent {
                     channelId={post.channel_id}
                     postId={post.id}
                     teamId={this.props.teamId}
-                    getDotMenuRef={this.getDotMenu}
+                    getDotMenuRef={this.refs.dotMenu}
```

Comment:
@vraravam I've found the issue causing the white screen @jgilliam17 mentioned. This needs to pass a function, otherwise `EmojiPickerOverlay` will fail here:
```
static getPlacement(props) {
        const target = props.target();
        if (target) {
            const targetBounds = target.getBoundingClientRect();
            return popOverOverlayPosition(targetBounds, window.innerHeight, props.spaceRequiredAbove, props.spaceRequiredBelow);
        }

        return 'top';
    }
```

I'm not sure if the other files are extraneous, but it does seem like a lot of changes for a TS migration. I couldn't find any reversions or anything, so I think we're actually okay there. If we can fix this then we should be okay going forward.

Reply:
thanks for confirming, but i dont have enough context to fix this finding...can @ashishbhate or anyone else please help?

Reply:
ok - fixing

Reply:
@devinbinnie - i think this will fix the issue.

Reply:
@vraravam Looks good, but now we have a type-check issue on the function call:
```
> mattermost-webapp@0.0.1 check-types /root/mattermost/mattermost-webapp
> tsc

components/post_view/post_info/post_info.tsx:216:35 - error TS2322: Type '() => React.ReactInstance' is not assignable to type 'Element | Component<any, {}, any> | undefined'.
  Type '() => ReactInstance' is missing the following properties from type 'Component<any, {}, any>': context, setState, forceUpdate, render, and 3 more.

216                     getDotMenuRef={this.getDotMenu}
                                      ~~~~~~~~~~~~~~~~~

  components/post_view/post_info/post_info.tsx:216:35
    216                     getDotMenuRef={this.getDotMenu}
                                          ~~~~~~~~~~~~~~~~~
    Did you mean to call this expression?


Found 1 error.
```

I think this can be fixed by changing the property type on `PostReaction` for `getDotMenuRef` to `() => HTMLDivElement`

When done, make sure to run a `make check-types` before you push to ensure that there are no more type issues.
Thanks again for the hard work! We're almost there!

Reply:
sorry @devinbinnie - last night was really late due to a 14 hour work day. I have now fixed the issue.

Reply:
@devinbinnie - this has been fixed approx 24 hrs ago and QA has also approved.